pkg/factory/risingwave_object_factory.go

```diff
@@ -1665,6 +1665,15 @@ func (f *RisingWaveObjectFactory) buildPodTemplateFromNodeGroup(component string
 		})
 	}
 
+	// set resource group in compute component.
+	if component == consts.ComponentCompute && nodeGroup.Name != "default" {
```

Comment:
there is a concern about open source operator user.
Their node group name may not be `default`. It will be a breaking change if they upgrade operator.
I don't have idea about it 🤔. cc @arkbriar

Reply:
> Their node group name may not be default. It will be a breaking change if they upgrade operator.
I don't have idea about it 🤔. cc @arkbriar

No worries, workload objects won't change immediately on operator upgrade. They will change on a further RW object spec change (which leads to a generate bump, [code](https://github.com/risingwavelabs/risingwave-operator/blob/7661afd22f705ff2886b3a35602e7e11e6c4ceef/pkg/manager/risingwave_controller_manager_impl.go#L709-L729)).

However, the default node group name is `""` (empty string). Please use that instead @CAJan93.

Reply:
> No worries, workload objects won't change immediately on operator upgrade. They will change on a further RW object spec change (which leads to a generate bump, [code](https://github.com/risingwavelabs/risingwave-operator/blob/7661afd22f705ff2886b3a35602e7e11e6c4ceef/pkg/manager/risingwave_controller_manager_impl.go#L709-L729)).

But it's still a breaking change right? When user updates an unrelated field after bumping operator, they don't expect that the resource group is also changed.

Besides, the `nodeGroup.Name` is a required field. I guess most user don't set it to empty string?

Reply:
> When user updates an unrelated field after bumping operator, they don't expect that the resource group is also changed.

Yes. But it only happens on non-default node groups. I don't think there will be much user using that.

> I guess most user don't set it to empty string?

The default value is "" if name is unset.